src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -206,15 +215,33 @@ private boolean areAllChildrenIgnored() {
     }
 
     /**
-     * Returns a {@link Statement}: run all non-overridden {@code @BeforeClass} methods on this class
+     * Returns a {@link Statement}: run all non-overridden {@code @Ignore} methods on this class
+     * and superclasses before executing {@code next}.
+     */
+    protected Statement withIgnores(final RunNotifier notifier) {
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                for (final T each : getFilteredChildren()) {
+                    if (isIgnored(each)) {
+                        ParentRunner.this.fireIgnoreChild(each, notifier);
+                    }
+                }
+            }
+        };
+    }
+
+    /**
+     * @param previous The previous statement, or {@code null}.
+     * @param next The next statement after methods annotated with {@link BeforeClass}.
+     * @return A {@link Statement}: run all non-overridden {@code @BeforeClass} methods on this class
      * and superclasses before executing {@code statement}; if any throws an
      * Exception, stop execution and pass the exception on.
      */
-    protected Statement withBeforeClasses(Statement statement) {
+    protected Statement withBeforeClasses(final Statement previous, final Statement next) {
```

Comment:
No problem, it can be provided as an overloaded method with one more parameter to leave the original one untouched. Final can be removed, too.